src/features/dsPreview/state/dsPreviewActions.ts

```diff
@@ -3,10 +3,15 @@ import { ApiAction, ApiActionThunk, CALL_API } from "../../../store/api";
 
 export function loadDsPreview(ref: QriRef): ApiActionThunk {
   return async (dispatch, getState) => {
-    return dispatch(fetchDsPreview(ref))
+    await Promise.all([dispatch(fetchDsPreview(ref))])
```

Comment:
This smells, but it gets the job done. We need to wait for the preview to load body and readme data (as it's not an all inclusive call) and they have to be added on to the preview.

Reply:
This is good for now. We can work on this on the backend